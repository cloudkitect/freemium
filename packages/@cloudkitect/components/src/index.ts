export * from './ck-certificatemanager/ck-dns-validated-certificate';
export * from './ck-cloudfront/ck-distribution';
export * from './ck-core/ck-vendor-tags';
export * from './ck-core/ck-tag';
export * from './ck-core/ck-utils';
export * from './ck-core/ck-region';
export * from './ck-dynamodb/ck-table-v2';
export * from './ck-ec2/ck-vpc';
export * from './ck-ecr/ck-repository';
export * from './ck-ecs/ck-cluster';
export * from './ck-ecs/ck-fargate-clsuter';
export * from './ck-ecs/ck-fargate-service';
export * from './ck-ecs/ck-fargate-task-definition';
export * from './ck-elasticloadbalancingv2/ck-public-application-loadbalancer';
export * from './ck-s3/ck-bucket';
export * from './ck-stack/ck-stack';
