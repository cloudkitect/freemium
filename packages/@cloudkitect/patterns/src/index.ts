export * from './ck-website/ck-serverless-website';
export * from './ck-container/ck-serverless-container-app';