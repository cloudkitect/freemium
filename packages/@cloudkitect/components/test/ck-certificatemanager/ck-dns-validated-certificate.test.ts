/*
 * Copyright (c) CloudKitect Inc. All Rights Reserved.
 */

import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { HostedZone, IHostedZone } from 'aws-cdk-lib/aws-route53';
import { CkAccountType, CkStack } from '../../lib';
import { CkDnsValidatedCertificate } from '../../src';

let app: App;
let stack: CkStack;
const domainName = 'example.com';
let hostedZone: IHostedZone;

const stackProps = {
  ckAccountType: CkAccountType.PRODUCTION,
  ckApplication: 'Test',
  ckCompany: 'CloudKitect',
};

describe('Well Architected DNS Validated certificate', () => {
  beforeEach(() => {
    app = new App();
    stack = new CkStack(app, 'Test-Stack', stackProps);
    hostedZone = new HostedZone(stack, 'HZone', {
      zoneName: domainName,
    });
  });

  test('Default dns certificate settings', () => {
    new CkDnsValidatedCertificate(stack, 'Certificate', {
      ckHostedZone: hostedZone,
      domainName,
    });

    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });


  test('CC-351: certificate validation via hosted zone', () => {
    new CkDnsValidatedCertificate(stack, 'Certificate', {
      domainName,
      ckHostedZone: hostedZone,
    });

    const template = Template.fromStack(stack);
    template.hasResourceProperties('AWS::CertificateManager::Certificate', {
      DomainValidationOptions: [
        {
          DomainName: 'example.com',
          HostedZoneId: {
            Ref: 'HZoneBA7FEF6F',
          },
        },
      ],
    });
  });
});
