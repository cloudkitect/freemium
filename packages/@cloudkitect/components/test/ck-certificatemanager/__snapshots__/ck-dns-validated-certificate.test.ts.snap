// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Well Architected DNS Validated certificate Default dns certificate settings 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Certificate4E7ABB08": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "example.com",
        "DomainValidationOptions": [
          {
            "DomainName": "example.com",
            "HostedZoneId": {
              "Ref": "HZoneBA7FEF6F",
            },
          },
        ],
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
          {
            "Key": "Name",
            "Value": "Test-Stack/Certificate",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "HZoneBA7FEF6F": {
      "Properties": {
        "Name": "example.com.",
      },
      "Type": "AWS::Route53::HostedZone",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
